src/DynamoCore/Properties/Resources.en-US.resx

```diff
@@ -908,4 +908,7 @@ This package likely contains an assembly that is blocked. You will need to load
   <data name="FormulaMigrated" xml:space="preserve">
     <value>Formula node has been deprecated. It has been automatically migrated to a CodeBlock node. Note that results may vary after the migration depending on lacing options selected on the original Formula node. Appropriate replication guides might need to be applied to the CodeBlock node script.</value>
   </data>
+  <data name="LegacyTraceDataWarning" xml:space="preserve">
+    <value>This workspace contains element binding data in a legacy format that is no longer supported in Dynamo 3.0 and higher versions. Element binding will work however, if any new trace data generated during graph execution is saved with the workspace.</value>
```

Comment:
What about:

`This workspace contains element binding data in a legacy format that is no longer supported in Dynamo 3.0 and higher versions. Element binding data will be saved in the new format the next time you save this workspace.`

?

Reply:
@mjkkirschner, question - what happens to the old soap formatted data when the workspace is executed and saved? Will the old trace data be overwritten? Is this tested?

Reply:
Actually it reminds me that we discussed (me you and @saintentropy) in our last discussion that it would be good to actually dump the old trace data - which is not being done in this PR as far as I can tell.

In the current state we're going to discard each callsite's data.

I think it would be preferable if in addition to showing the warning, you discard the trace data. IE if you need to show a warning, then the trace data is not useful and should be dumped.

Reply:
And to answer your question more specifically, no, I don't think we have a test for this specifically.

Reply:
We could deal with this optimization as a separate PR/task if you want to keep it isolated from the UI changes - but I guess it will touch the same check in the deserialization logic.

Reply:
Ok, will keep this PR open and work on discarding the old data here as well.